subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/executer/InProcessGradleExecuter.java

```diff
@@ -279,6 +280,7 @@ private BuildResult executeBuild(GradleInvocation invocation, StandardOutputList
             BuildActionParameters buildActionParameters = createBuildActionParameters(startParameter);
             BuildRequestContext buildRequestContext = createBuildRequestContext(outputListener, errorListener);
             startMeasurement();
+            GLOBAL_SERVICES.get(MemoryStatusBroadcaster.class).start();
```

Comment:
The broadcaster should start itself, or at least be started by some global service.

Having the entry points know which internal things they need to start (and stop) is not particularly robust. For example, we've already missed an entry point (tooling api embedded mode).